src/Tickets/Emails/Dispatcher.php

```diff
@@ -0,0 +1,507 @@
+
+/**
+ * Class Dispatcher.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Emails
+ */
+class Dispatcher {
+
+	/**
+	 * Whether this dispatcher was used.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected bool $used = false;
+
+	/**
+	 * Stores the Email used in this Dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @var Email_Abstract|null
+	 */
+	protected ?Email_Abstract $email = null;
+
+	/**
+	 * Stores the all the headers used on the email.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,string>
+	 */
+	protected array $headers = [];
+
+	/**
+	 * Stores the all the attachments used on the email.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	protected array $attachments = [];
+
+	/**
+	 * Stores the contents to be dispatched.
+	 *
+	 * @since TBD
+	 *
+	 * @var ?string
+	 */
+	protected ?string $content = null;
+
+	/**
+	 * Stores where we will dispatch to.
+	 *
+	 * @since TBD
+	 *
+	 * @var ?string
+	 */
+	protected ?string $to = null;
+
+	/**
+	 * Stores the subject to be dispatched.
+	 *
+	 * @since TBD
+	 *
+	 * @var ?string
+	 */
+	protected ?string $subject = null;
+
+	/**
+	 * Sets the Email instance that will be used for this Dispatcher, intentionally this is protected method.
+	 * You can only generate a new Dispatcher from the Factory Method.
+	 *
+	 * @since TBD
+	 *
+	 * @param Email_Abstract $email
+	 */
+	protected function set_email( Email_Abstract $email ): void {
+		$this->email = $email;
+	}
+
+	/**
+	 * Gets the email instance for this Dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @return Email_Abstract|null
+	 */
+	public function get_email(): ?Email_Abstract {
+		return $this->email;
+	}
+
+	/**
+	 * Prepares the current dispatcher for sending an email.
+	 *
+	 * @since TBD
+	 *
+	 * @param Email_Abstract $email Email Instance used to prepare the dispatcher with.
+	 */
+	protected function prepare_dispatcher( Email_Abstract $email ): void {
+		// Enforce text/html content type header.
+		$this->add_header( 'Content-Type', 'text/html; charset=utf-8' );
+		$this->set_email( $email );
+
+		$from_email = $email->get_from_email();
+		$from_name  = $email->get_from_name();
+
+		// Add From name/email to headers if no headers set yet, and we have a valid From email address.
+		if ( ! empty( $from_name ) && ! empty( $from_email ) && is_email( $from_email ) ) {
+			$from_email = sanitize_email( $from_email );
+
+			$this->add_header( 'From', sprintf(
+				'%1$s <%2$s>',
+				stripcslashes( $from_name ),
+				$from_email
+			) );
+
+			$this->add_header( 'Reply-To', $from_email );
+		}
+
+		$this->set_to( $email->get_recipient() );
+		$this->set_subject( $email->get_subject() );
+		$this->set_content( $email->get_content() );
+	}
+
+	/**
+	 * From a given email type instance generate a new Dispatcher and prepare it for usage.
+	 *
+	 * @since TBD
+	 *
+	 * @param Email_Abstract $email
+	 *
+	 * @return Dispatcher
+	 */
+	public static function from_email( Email_Abstract $email ): Dispatcher {
```

Comment:
This makes it so confusing. The `from` word is pretty much reserved for other types of usages in the "emails" world.

I think for the sake of clarity this method should be renamed.

Reply:
This is marking that we are generating a dispatcher from a Email Type instance. It's a factory method.